gegelatilib/include/data/demangle.h

```diff
@@ -0,0 +1,18 @@
+#ifndef GEGELATI_DEMANGLE_H
+#define GEGELATI_DEMANGLE_H
+
+#include <string>
+
+#ifdef _MSC_VER
+/// Macro for getting type name in human readable format.
+#define DEMANGLE_TYPEID_NAME(name) name
+#elif __GNUC__
+#include <cxxabi.h>
+std::string demangle(const char* name);
```

Comment:
Since the h file is in the `data` folder, the function should be part of the `Data` namespace.